src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -870,7 +872,19 @@ private void ScanSyntaxToken(ref TokenInfo info)
                         TextWindow.AdvanceChar();
                     }
 
-                    info.Text = TextWindow.GetText(intern: true);
+                    if (_errorCount++ > 100)
```

Comment:
@sharwell Re "What about the special error message if you hit the limit?"

We are past the cutoff for adding new diagnostics in VS 2015. The set of messages has been frozen for translation.